pyproject.toml

```diff
@@ -103,3 +104,29 @@ extend-ignore = [
   "W503",  # line break before binary operator
   "W504",  # line break after binary operator
 ]
+
+[tool.mypy]
+python_version = "3.10"
+warn_return_any = true
+warn_unused_configs = true
+# These are disabled during gradual typing migration
+# disallow_untyped_defs = true
+# disallow_incomplete_defs = true
+check_untyped_defs = true
+# disallow_untyped_decorators = true
+no_implicit_optional = true
+warn_redundant_casts = true
+warn_unused_ignores = false
+warn_no_return = true
+warn_unreachable = true
+strict_equality = true
+extra_checks = true
+
+[[tool.mypy.overrides]]
+module = [
```

Comment:
Please check the entire list thoroughly. I found that argcomplete also provides type annotations.